FW/src/Drivers/LIS3DH.cpp

```diff
@@ -144,69 +147,100 @@ namespace Accelerometer
 
     static const uint8_t WhoAmI = 0b00110011;
 
-    LIS3DH::LIS3DH(I2C::I2CDevice *i2c, GPIO::Pin *intPin) {
-        this->m_i2c = i2c;
-        this->m_intPin = intPin;
-        this->m_initialized = false;
+    LIS3DH::LIS3DH(I2C::I2CDevice *i2c, GPIO::Pin *intPin, FreeRTOS::Queue<Events, EventQueueLength> *eventQueue) : 
+        Task("Acc", Task::TaskPriority::MediumPriority) {
+        m_i2c = i2c;
+        m_intPin = intPin;
+        m_initialized = false;
+        m_eventQueue = eventQueue;
     }
 
 
-    void LIS3DH::Init() {
+    void LIS3DH::Run() {
         uint8_t whoami;
 
-        this->m_i2c->ReadReg(WHO_AM_I, &whoami);
+       m_i2c->ReadReg(WHO_AM_I, &whoami);
 
-        if(whoami != WhoAmI) {
+        if (whoami != WhoAmI)
```

Comment:
Hmm, as WhoAmI is the answer, maybe call it WhoIAm? =)

Reply:
It is named after a register in the accelerometer:
https://www.st.com/resource/en/datasheet/lis3dh.pdf

Chapter 8.5